workbench-core/accounts/src/services/projectService.ts

```diff
@@ -127,91 +141,58 @@ export default class ProjectService {
       return { data: [project], paginationToken: undefined };
     }
 
-    // Else, member of more than 1 group, query for project items with pagination
-    // welcome to party town
+    // Else, member of more than 1 group, batch get items and manually filter, sort, and paginate
     const projectIds: string[] = userGroupsForCurrentUser.map(
       (projectGroup: string) => projectGroup.split('#')[0]
     );
-
-    // i = 0
-    let filterExpression = 'id = :proj1';
-    const expressionAttributeValues: { [key: string]: string } = { ':proj1': projectIds[0] };
-    for (let i = 1; i < projectIds.length; i++) {
-      // build filter string "id = :proj1 OR id = :proj2 OR ... OR id = projN+1"
-      filterExpression = filterExpression + ` OR id = :proj${i + 1}`;
-      // build expressionAttributeValues object {":proj1": "projectId1", ..., ":projN+1": "projectIdN"}
-      expressionAttributeValues[`:proj${i + 1}`] = projectIds[i];
-    }
-
-    // TODO--extend to other GSIs after filtering is added
-    let queryParams: QueryParams = {
-      key: { name: 'resourceType', value: this._resourceType },
-      index: 'getResourceByCreatedAt',
-      limit: pageSize,
-      filter: filterExpression,
-      values: expressionAttributeValues
-    };
-
-    let projects: Project[] = [];
-
-    let returnedCount = 0;
-    while (returnedCount < pageSize && returnedCount < projectIds.length) {
-      queryParams = addPaginationToken(paginationToken, queryParams);
-
-      const projectsResponse = await this._aws.helpers.ddb.query(queryParams).execute();
-
-      returnedCount += projectsResponse.Count!;
-
-      // nothing returned
-      if (projectsResponse.Items === undefined || _.isEmpty(projectsResponse.Items)) {
-        paginationToken = getPaginationToken(projectsResponse);
-        console.log(paginationToken);
-      }
-
-      // nothing else to get from DDB
-      else if (projectsResponse.LastEvaluatedKey === undefined) {
-        projects = projects.concat(projectsResponse.Items.map((item) => this._formatFromDDB(item)));
-        paginationToken = undefined;
+    const keys: { [key: string]: unknown }[] = projectIds.map((projectId) =>
+      buildDynamoDBPkSk(projectId, resourceTypeToKey.project)
+    );
+    const projectsResponse = (await this._aws.helpers.ddb.get(keys).execute()) as BatchGetItemCommandOutput;
+    if (projectsResponse.Responses) {
+      const projects: Project[] = projectsResponse.Responses[this._tableName].map((item) =>
+        this._formatFromDDB(item)
+      );
+      // apply sort or filter
```

Comment:
If I made this a separate function, it would be private so then we could not unit test it. Since this is just one line essentially, I am going to keep it like this.

Reply:
You wouldn't make it a private method, you'd make it an object whose responsibility is filtering and sorting. The methods would be public on that object.